### PR TITLE
Add proper cfgs for struct HirIdValidator used only with debug-assert

### PR DESCRIPTION
**Pull request by mu001999**
See https://github.com/rust-lang/rust/pull/122065#issuecomment-1980118572.
I think it's due to #121752.

### CODE REVIEW

File: compiler/rustc_passes/src/hir_id_validator.rs [hunk 1]
```diff
@@ -1,11 +1,13 @@
...
+    rustc_data_structures::sync::Lock,
+    rustc_hir as hir,
+    rustc_hir::def_id::LocalDefId,
+    rustc_hir::{intravisit, HirId, ItemLocalId},
+    rustc_index::bit_set::GrowableBitSet,
+    rustc_middle::hir::nested_filter,
+};
 
 pub fn check_crate(tcx: TyCtxt<'_>) {
     if tcx.sess.opts.unstable_opts.hir_stats {
```
Comment (oli-obk) [inline compiler/rustc_passes/src/hir_id_validator.rs:13]: maybe inline this one at the one call site and then cfg out the entire module plus the `check_crate` call site?

Comment (oli-obk) [reply compiler/rustc_passes/src/hir_id_validator.rs]: Sorry, I meant cfging the `mod` declaration in the parent module, not adding a nested module.

**Comment by oli-obk** [comment]
Thanks!

@bors r+ rollup

**Comment by matthiaskrgr** [comment]
@bors r-
https://github.com/rust-lang/rust/pull/122096#issuecomment-1981833485

**Comment by oli-obk** [comment]
probably some cfg shenanigans going on in rustdoc. try rerunning with `MAGIC_EXTRA_RUSTFLAGS=-Ztreat-err-as-bug` to get a backtrace of where the error occurs, maybe something obvious stands out

cc @rust-lang/rustdoc any idea what's up here? rustdoc is trying to resolve a function call in cfged out code

**Comment by Nemo157** [comment]
Maybe related to usage of `--generate-link-to-definition` and this FIXME

https://github.com/rust-lang/rust/blob/1508a031fbfa55d11f1e30b2cb4d956554937f78/src/librustdoc/html/render/span_map.rs#L161-L167

**Comment by oli-obk** [comment]
@mu001999 you can remove the `--generate-link-to-definition` flags from bootstrap until that FIXME is resolved.

**Pull request merged**